### ISSUE TITLE
prov/verbs;ofi_rxm Initiating a data transfer seg faults in rxm_ep_rma_common() at line 308

### ISSUE THREAD

**Issue by tonyzinger** [ofiwg/libfabric#4190]
When doing a RMA transfer using fi_write(), it seg faults in rxm_ep_rma_common() at line 308.
When doing a Message using fi_send() or a Tagged Message using fi_tsend(), it seg faults in rxm_ep_send_common() at line 1089.

It appears that the handles_av entries in the rxm_ep->util_ep.cmap are NULL.

I am including as supporting documentation, gdb information that I have collected from a dump for the RMA test case.  Also, I am including the debug output from my test case.

[dump_info.txt](https://github.com/ofiwg/libfabric/files/2099454/dump_info.txt)
[RMA_verbs_ofi_rxm_rank_0.txt](https://github.com/ofiwg/libfabric/files/2099455/RMA_verbs_ofi_rxm_rank_0.txt)

**Comment by jswaro** [comment]
@shefty @gladkovdmitry17  

I've been looking over some of this with Tony but I'm not seeing anything that jumps out to me other than he requested a table and seems to have a map (when I looked at the core dump).

**Comment by dmitrygladkov** [comment]
```
[Rank:  0] DEBUG: ltu_av_init_mt: fi_ep_bind has been successful
[Rank:  0] DEBUG: Address Vector has been successfully opened
[Rank:  0] INFO: local address: 'fi_sockaddr_in://10.100.49.6:38087', address_length: 35
[Rank:  0] INFO: [scatter index: 0] id: 0, address: 'fi_sockaddr_in://10.100.49.6:38087', address_length: 35
[Rank:  0] INFO: [gather index: 0] id: 0, address: 'fi_sockaddr_in://10.100.49.6:38087', address_length: 35
[Rank:  0] INFO: [gather index: 1] id: 1, address: 'fi_sockaddr_in://10.100.49.7:53275', address_length: 35
[Rank:  0] INFO: [gather index: 2] id: 2, address: 'fi_sockaddr_in://10.100.49.8:58657', address_length: 35
[Rank:  0] INFO: [sorted gather index: 0] address: 'fi_sockaddr_in://10.100.49.6:38087', address_length: 35
[Rank:  0] INFO: [sorted gather index: 1] address: 'fi_sockaddr_in://10.100.49.7:53275', address_length: 35
[Rank:  0] INFO: [sorted gather index: 2] address: 'fi_sockaddr_in://10.100.49.8:58657', address_length: 35
[Rank:  0] DEBUG: AV addresses have been gathered successfully
[Rank:  0] INFO: [insert index: 0] address: 'fi_sockaddr_in://10.100.49.6:38087', address_length: 35
[Rank:  0] INFO: [insert index: 1] address: 'fi_sockaddr_in://10.100.49.7:53275', address_length: 35
[Rank:  0] INFO: [insert index: 2] address: 'fi_sockaddr_in://10.100.49.8:58657', address_length: 35
**[Rank:  0] DEBUG: fi_av_insert has been successful**
**[Rank:  0] DEBUG: endpoints have been successfully enabled**
[Rank:  0] DEBUG: AV addresses have been successfully inserted
[Rank:  0] DEBUG: ltu_insert_av_addresses has been successful
[Rank:  0] DEBUG: addresses have been successfully initialized
```
There are two ways to fix the issue:
- move an allocation of a connection map from fi_enable(ep) call to fi_ep_bind(av, ep) -- this allows an insertion of addresses into AV before EP is turned to ENABLE state. Feel free to try this fix from `cmap` branch in my fork ([dmitrygladkov/cmap](https://github.com/dmitrygladkov/libfabric/tree/cmap))
- change your application sequence in your application to insert addresses into AV after EP is enabled.

@shefty What is a more proper way to resolve the issue?

**Comment by shefty** [comment]
The man pages don't document that there's any AV <-> EP state that needs to be maintained prior to inserting addresses into the AV.  I can envision an application inserting addresses into an AV even prior to the EP being associated with it.  So, my current preference is to provide a fix for the problem within the provider and not require the application to change.

But, I'm going to look into the AV implementations quickly to see if there's justification for requiring that an AV be in a specific state relative to the associated EPs prior to inserting addresses.

**Comment by dmitrygladkov** [comment]
@shefty Thanks! Let me know if you find anything. I'll prepare a PR to fix this issue.
@jswaro @tonyzinger Could you test the possible fix from my GH fork, please?

**Comment by jswaro** [comment]
@dmitrygladkov We can do that.

**Comment by jswaro** [comment]
@dmitrygladkov Did you have a PR for us to look at?

**Comment by dmitrygladkov** [comment]
Sorry, I'm waiting the results of test from you.


> (dmitrygladkov/cmap)

I suggested to verify this fix. I'll prepare a PR as soon as you confirm that this works for you.

**Comment by jswaro** [comment]
@dmitrygladkov Sure, but what branch contains the fix you mentioned above?

**Comment by dmitrygladkov** [comment]
@jswaro Right, this is cmap branch - https://github.com/dmitrygladkov/libfabric/tree/cmap

**Comment by jswaro** [comment]
Ok, We'll get that tested and let you know. Thanks!

**Comment by tonyzinger** [comment]
Dmitry's fix has solved this problem.  I am now able to do some data transfers successfully.

**Comment by dmitrygladkov** [comment]
> Dmitry's fix has solved this problem. I am now able to do some data transfers successfully.

Thanks for testing this. I'll prepare an appropriate PR with those changes soon.
